src/Util/ClassSourceManipulator.php

```diff
@@ -438,16 +508,21 @@ private function addSingularRelation(BaseRelation $relation)
                 'nullable' => false,
             ]);
         }
+
         $this->addProperty($relation->getPropertyName(), $annotations);
 
         $this->addGetter(
             $relation->getPropertyName(),
-            $typeHint,
+            $relation->getCustomReturnType() ?: $typeHint,
             // getter methods always have nullable return values
-            // because even though these are required in the db, they may not be set
-            true
+            // unless this has been customized explicitly
+            $relation->isCustomReturnTypeNullable()
```

Comment:
I still think this should be:

```php
$relation->getCustomReturnType() ? $relation-> isCustomReturnTypeNullable() : true
```

It, of course, makes no technical difference. The difference is in how the code "reads". Your code says:

> Always read the custom return type nullable value... which happens to default to true

Mine says:

> If the user has customized the return type, use their preference for nullable vs not nullable. If they have
> *not* customized the return type, then we know that, in this situation, the return type should default to `true`.

Reply:
Done.